feat: global tags, upgrade to influxdb, upgrade to telegraph

sstarcher opened the pull request:
This contains the following
- Upgrade to InfluxDB 0.11.1 - I was causing crashes in 0.10.x
- Upgrade to Telegraf 0.11.1 - 0.10.x was possibly causing leaked sockets
- Upgrade to Alpine 3.3
- Conversion of envtpl for telegraf to dockerize - this was done to add support for global tag as an environment variable
- Optional disabling of Prometheus integration -  I normally just rebuild and delete the code related to this, but this will allow me to turn it off and make no difference on the Deis side.
- Stripe duplicate blanks lines from the config template that is generated.  - Makes it easier to read

sstarcher commented:
@jchauncey this is part of my stab at  #78

jchauncey commented on telegraf/rootfs/Dockerfile:
```diff
@@ -1,20 +1,22 @@
+RUN \
+    mkdir -p /usr/local/bin/ &&\
+    apk --update add curl &&\
+    curl -SL https://github.com/jwilder/dockerize/releases/download/v${DOCKERIZE_VERSION}/dockerize-linux-amd64-v${DOCKERIZE_VERSION}.tar.gz \
```
whats the point of this package?

sstarcher replied:
string splitting as I did not realize envtpl had sprig due to it's non-existent documentation

jchauncey commented on telegraf/rootfs/config.toml.tpl:
```diff
@@ -1,99 +1,103 @@
+    {{ range $index, $element := split .Env.GLOBAL_TAGS "," }}
+      {{ $value := split $element ":" }}{{index $value 0}}="{{index $value 1}}"
+    {{end}}
+{{ end }}
```
you could have accomplished this same thing with envtpl which includes [sprig](https://github.com/Masterminds/sprig)

sstarcher replied:
that would of saved me a lot of time

jchauncey commented:
All you needed to do was add your range section to the template and that would do exactly what you wanted without having to change the entire template. Also if you turn off the prometheus part you wont receive any metrics from kubernetes which we are building dashboards for.

sstarcher commented:
@jchauncey I don't currently run prometheus.  I may in the future or when your dashboard exists.  I'm using heapster to collect my kubernetes metrics.  This weekend I can revert all of those changes to the template and use sprig instead.  Envtpl has no documentation and I had no clue it included sprig.

jchauncey commented:
No you dont need prometheus. We are gathering the kubernetes metrics which are exposed as prometheus endpoints and sending them to influx.

Its exactly how cadvisor+heapster works but this way we have all the metrics/graphs in 1 spot

sstarcher commented:
Not all of my nodes are kubernetes nodes

jchauncey commented:
What do you mean? This project is kind of built to gather metrics from a kubernetes cluster.

sstarcher commented:
@jchauncey I can certainly fork your stuff if you prefer to not add the configuration option.  I mean I run both a kubernetes cluster and a few nodes that are not part of the kubernetes cluster, but are part of the same fleet cluster.

So far as long as the prometheus integration configurable I can use everything else.

sstarcher commented:
I also currently run a Deis cluster and am looking forward to the 2.0 release.

jchauncey commented:
Well this particular component is meant to provide a robust metrics system for deis running on k8s so thats kind of the base. I like the global tags part of the PR just wonder about making the prometheus portion configurable. 

Basically telegraf will only run on your kubernetes cluster anyways when you install this (using the monitor chart we are building).

jchauncey commented:
in the case were you want to collect metrics from non-k8s nodes you could just run the official docker image of telegraf.

sstarcher commented:
annnnnd I just accidently wiped out all my changes...  give me a few moments

sstarcher committed:
feat(telegraf/template): adds global tags to telegrap template

global tags, upgrade to influxdb, upgrade to telegraph, turns off k8s integration when not in k8s

sstarcher commented:
@jchauncey converted everything back to envtpl and used sprig.
I changed the disabling of the k8s by checking if the file exists instead of a flag.

There is no official telegraf docker container.  Along with anything I contribute here I would have to reproduce like the global_tags in this pull request.  I would prefer not to run 2 seperate telegraf setups, but I will if I must it's just less desirable.

jchauncey commented:
K let me think over the weekend about how to best approach this. I never really considered running this outside of k8s so it may mean rethinking stuff. but there is an official image now - https://github.com/influxdata/telegraf-docker

sstarcher commented:
ya an official image using trusty, before using that I would just fork this project.

jchauncey commented:
Ok im liking what I see here. @sstarcher could you format your commit message using these guidelines - https://github.com/deis/workflow/blob/master/src/contributing/submitting-a-pull-request.md

Once you do that we can work on getting this merged.

sstarcher commented:
@jchauncey I reformatted the commit message
